packages/core/src/renderer/components/+namespaces/subnamespace-badge.tsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import styles from "./subnamespace-badge.module.scss";
+
+import React from "react";
+import type { Namespace } from "../../../common/k8s-api/endpoints";
+import { Tooltip } from "../tooltip";
+
+interface SubnamespaceBadgeProps extends React.HTMLAttributes<HTMLSpanElement> {
+  namespace: Namespace;
+  id: string;
+}
+
+export function SubnamespaceBadge({ namespace, id, ...other }: SubnamespaceBadgeProps) {
+  if (!namespace.getAnnotations().find(annotation => annotation.includes("hnc.x-k8s.io/subnamespace-of"))) {
```

Comment:
Probably better to create `Namespace.isSubnamespace(){  return this.getAnnotations().find(annotation => annotation.includes("hnc.x-k8s.io/subnamespace-of")) }` method and re-use where it needed.